sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/PushDownUtils.scala

```diff
@@ -70,6 +72,36 @@ object PushDownUtils extends PredicateHelper {
     }
   }
 
+  /**
+   * Pushes down aggregates to the data source reader
+   *
+   * @return pushed aggregation.
+   */
+  def pushAggregates(
+      scanBuilder: ScanBuilder,
+      aggregates: Seq[AggregateExpression],
+      groupBy: Seq[Expression]): Aggregation = {
+
+    def columnAsString(e: Expression): Option[FieldReference] = e match {
+      case AttributeReference(name, _, _, _) => Some(FieldReference(Seq(name)))
+      case _ => None
+    }
+
+    scanBuilder match {
+      case r: SupportsPushDownAggregates =>
+        val translatedAggregates = aggregates.map(DataSourceStrategy.translateAggregate)
+        val translatedGroupBys = groupBy.map(columnAsString)
+
+        val agg = Aggregation(translatedAggregates.flatten, translatedGroupBys.flatten)
```

Comment:
I think we can only apply pushdown if all the group by cols are supported. e.g. `GROUP BY a, substring(b), c`, it's wrong to pushdown `GROUP BY a, c`